modelerfour/src/quality-precheck/prechecker.ts

```diff
@@ -336,6 +299,60 @@ export class QualityPreChecker {
     return undefined;
   }
 
+  private findSchemaToRemove(
+    schema1: DereferencedSchema,
+    schema2: DereferencedSchema,
+  ): { keep: DereferencedSchema; remove: DereferencedSchema } {
+    const schema1Ref = this.input.components?.schemas?.[schema1.key].$ref;
+    // If schema1 is pointing to schema2 then we should delete schema1
+    if (schema1Ref && schema1Ref === `#/components/schemas/${schema2.key}`) {
+      return { remove: schema1, keep: schema2 };
+    }
+    return { remove: schema2, keep: schema1 };
+  }
+
+  private removeDuplicateSchemas(name: string, schema1: DereferencedSchema, schema2: DereferencedSchema) {
+    const { keep: schemaToKeep, remove: schemaToRemove} = this.findSchemaToRemove(schema1, schema2);
+    delete this.input.components!.schemas![schemaToRemove.key];
+    const text = JSON.stringify(this.input);
+    this.input = JSON.parse(
+      text.replace(
+        new RegExp(`"\\#\\/components\\/schemas\\/${schemaToRemove.key}"`, "g"),
```

Comment:
Should we cache this regex or does V8 do that internally?

Reply:
well the regex needs the key to change so it shouldn't really be used multiple times

Reply:
Ahhh, yeah, good point